src/main/src/main/java/org/geoserver/config/GeoServerDataDirectory.java

```diff
@@ -85,7 +85,38 @@ public class GeoServerDataDirectory {
     /** resource loader */
     GeoServerResourceLoader resourceLoader;
 
-    EntityResolverProvider entityResolverProvider;
+    /**
+     * Static reference to the EntityResolverProvider to avoid circular dependency issues during startup.
+     *
+     * <p>This static field prevents deadlocks that can occur when accessing the EntityResolverProvider through
+     * GeoServerExtensions during catalog loading (particularly for style group layer groups), as Spring might still be
+     * initializing beans.
+     *
+     * <p>The field must be static because some code (like ResourcePool.dataDir()) directly instantiates a
+     * GeoServerDataDirectory instead of relying on the Spring bean of that type.
+     *
+     * <p>The deadlock occurs in the following call chain:
+     *
+     * <pre>
+     * loadCatalog() ->
+     * CatalogImpl.add(LayerGroup) ->
+     *  validate(LayerGroup) ->
+     *   StyledLayerDescriptor sld = styles.get(i).getSLD() ->
+     *    StyleInfoImpl.getSLD() ->
+     *     ResourcePool.getSld(StyleInfo) ->
+     *      GeoServerDataDirectory.parseSld(StyleInfo) ->
+     *       GeoServerDataDirectory.getEntityResolver() ->
+     *        GeoServerExtensions.bean(EntityResolverProvider.class) ->
+     *         DefaultListableBeanFactory.getSingletonFactoryBeanForTypeCheck()
+     * </pre>
```

Comment:
Use of code to avoid ``-&lt;`` arrows:

```suggestion
     * <pre>
     * {@code
     * loadCatalog() ->
     * CatalogImpl.add(LayerGroup) ->
     *  validate(LayerGroup) ->
     *   StyledLayerDescriptor sld = styles.get(i).getSLD() ->
     *    StyleInfoImpl.getSLD() ->
     *     ResourcePool.getSld(StyleInfo) ->
     *      GeoServerDataDirectory.parseSld(StyleInfo) ->
     *       GeoServerDataDirectory.getEntityResolver() ->
     *        GeoServerExtensions.bean(EntityResolverProvider.class) ->
     *         DefaultListableBeanFactory.getSingletonFactoryBeanForTypeCheck()
     * }
     * </pre>
```

I java 18 you can use @snippet or something.